src/lm_buddy/jobs/_entrypoints/prometheus.py

```diff
@@ -0,0 +1,131 @@
+# lm-buddy entrypoint to run evaluations using a Prometheus inference server
+# see https://github.com/kaistAI/prometheus/blob/main/evaluation/benchmark/run_absolute_scoring.py
+
+from lm_buddy.jobs.configs import PrometheusJobConfig
+from lm_buddy.integrations.huggingface import HuggingFaceAssetLoader
+from lm_buddy.integrations.wandb import (
+    ArtifactType, 
+    ArtifactLoader, 
+    build_file_artifact, 
+    wandb_init_from_config,
+)
+from fastchat.conversation import get_conv_template
+from transformers import AutoTokenizer
+from openai import OpenAIError, OpenAI
+
+from tqdm import tqdm
+import os
+import json
+import copy
+
+class BadResponseException(Exception):
+    def __init__(self, message, error):
+        self.message = message
+        self.error = error
+
+
+def openai_completion(config, client, prompt):
+    return client.completions.create(
+        model = "kaist-ai/prometheus-13b-v1.0",
+        prompt = prompt,
+        best_of = config.prometheus.best_of,
+        max_tokens = config.prometheus.max_tokens,
+        frequency_penalty = config.prometheus.frequency_penalty,
+        temperature = config.prometheus.temperature,
+        top_p = config.prometheus.top_p
+    )
+
+
+def parse_response(response):
+    try:
+        assert response is not None
+        response_text = response.choices[0].text
+        feedback, score = response_text.split('[RESULT]')
+        feedback = feedback.strip()
+        score = score.strip()
+        assert score in ["1","2","3","4","5"]
+    except (ValueError, AssertionError) as e:
+        raise BadResponseException("Server returned a bad response", e)
+
+    return feedback, score
+
+
+def instruction_to_prompt(instruction):
+    conv = get_conv_template("llama-2")
+    conv.set_system_message("You are a fair evaluator language model.")
+    conv.append_message(conv.roles[0], instruction)
+    conv.append_message(conv.roles[1], None)
+    return conv.get_prompt()
+
+
+def run_prometheus(config: PrometheusJobConfig, artifact_loader: ArtifactLoader):
+
+    # load dataset from W&B artifact
+    hf_loader = HuggingFaceAssetLoader(artifact_loader)
+    artifact_path,_ = hf_loader.resolve_asset_path(config.dataset.load_from)
+    dataset_fname = os.path.join(artifact_path, config.dataset.load_from.name)
+    
+    with open(dataset_fname,'r') as f:
+        # eval samples are JSON-encoded, each takes one line in the dataset file
+        data = [json.loads(line) for line in f.readlines()]
+
+    # get the tokenizer
+    tokenizer = hf_loader.load_pretrained_tokenizer(config.prometheus.tokenizer)
+
+    # instantiate OpenAI client to speak with the vLLM endpoint
+    client = OpenAI(
+        base_url = config.prometheus.inference.base_url
+    )
+
+    # open the output file for writing and iterate on samples
+    output_fname = os.path.join("/tmp", config.tracking.name)
+    with open(output_fname,'w') as file:
+        for sample in tqdm(data):
```

Comment:
does tqdm work the same way in ray logs as regularly? might be something to check here if we're planning on displaying this output in the Ray Dashboard logs

Reply:
Yesterday was all local testing, ray is next!
My intuition is it will be a bit ugly (as it usually happens with other tqdm-based iterations such as loading models or trainings) but should not break stuff.
I'll also make sure the logging is a bit less verbose than now (e.g. I could skip printing all the answers, which are on rn for debugging in terminal)

Reply:
FWIW, in the finetuning code I set `disable_tqdm=True` in the training args because I think it shows up malformatted in the running logs.